src/Npgsql.GeoJSON/GeoJSONHandler.cs

```diff
@@ -0,0 +1,792 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2018 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using GeoJSON.Net;
+using GeoJSON.Net.CoordinateReferenceSystem;
+using GeoJSON.Net.Geometry;
+using Npgsql.BackendMessages;
+using Npgsql.TypeHandling;
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+namespace Npgsql.GeoJSON
+{
+    [Flags]
+    public enum GeoJSONOption
+    {
+        None = 0,
+        BoundingBox = 1,
+        ShortCrs = 2,
+        LongCrs = 4
+    }
+
+    public class GeoJSONHandlerFactory : NpgsqlTypeHandlerFactory<GeoJSONObject>
+    {
+        internal readonly GeoJSONOption _option;
+
+        public GeoJSONHandlerFactory(GeoJSONOption option = GeoJSONOption.None)
+            => _option = option;
+
+        protected override NpgsqlTypeHandler<GeoJSONObject> Create(NpgsqlConnection conn)
+        {
+            var builder = new CrsMapBuilder();
+            if ((_option & (GeoJSONOption.ShortCrs | GeoJSONOption.LongCrs)) != GeoJSONOption.None)
+                using (var cmd = new NpgsqlCommand(
+                    "SELECT min(srid), max(srid), auth_name " +
```

Comment:
This SQL gets executed every time a physical connection is made. This isn't the end of the world (because it's per physical connection, not pooled), but you may still want to consider a cache based on the connection string to load it only once. Although I see that you load from database only if special options are specified, so it may not be worth it etc.

Reply:
Yes, but there are a few thousand records. Fetching them each time will hurt performance.